src/main/java/core/org/akaza/openclinica/domain/datamap/EventCrf.java

```diff
@@ -356,5 +359,47 @@ public FormLayout getFormLayout() {
     public void setFormLayout(FormLayout formLayout) {
         this.formLayout = formLayout;
     }
+   
+
+	public static Comparator<EventCrf> getCompareByOrdinal() {
+		if(compareByOrdinal != null) {
+			return compareByOrdinal;
+		}else {
+			compareByOrdinal = new Comparator<EventCrf>() {
+			    @Override
+			    public int compare(EventCrf o1, EventCrf o2) {
+			    	Integer o1ordinal,o2ordinal;
+			    	o1ordinal = getOrdinal(o1);
+			    	o2ordinal = getOrdinal(o2);
+			        return o1ordinal.compareTo(o1ordinal);
+			    };
+                 
+			    /**
+				 * @param ec
+				 * @return 
+				 */
+				private Integer getOrdinal(EventCrf ec) {
+					Integer ecOrdinal = null;
+					List<EventDefinitionCrf> edfcList1 = ec.getFormLayout().getCrf().getEventDefinitionCrfs();
```

Comment:
Let's test this logic with two events, Event1 & Event2 with forms 1, 2 & 3 with different ordering in Event1 vs Event2.

Reply:
done